controllers/numaresourcesoperator_controller.go

```diff
@@ -305,6 +311,53 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
 	return daemonSetsNName, nil
 }
 
+func (r *NUMAResourcesOperatorReconciler) deleteUnusedDaemonSets(ctx context.Context, instance *nropv1alpha1.NUMAResourcesOperator, mcps []*machineconfigv1.MachineConfigPool) error {
+	klog.Info("Delete DS start")
+	var daemonSetList appsv1.DaemonSetList
+	if err := r.List(ctx, &daemonSetList, &client.ListOptions{Namespace: instance.Namespace}); err != nil {
+		klog.Infof("error while getting DS list: %v", err)
+		return err
+	}
+
+	// Generate the names of the DaemonSets that should be running,
+	// one for each MachineConfigPool
+	expectedDaemonSetNames := map[string]struct{}{}
```

Comment:
you may want to use `sets.String` from the k8s apimachinery package (we should have examples around)

Reply:
An example https://github.com/k8stopologyawareschedwg/resource-topology-exporter/blob/2a8d8ac49582d5b32bcfedb87341b3f4d6315c03/pkg/resourcemonitor/resourcemonitor.go#L61

Reply:
done